test : Manual testing POST /register

Aryamanz29 opened the pull request:
### Description

- Perform manual testing and report `POST /register API`
-  Added Test Report  Markdown `.\docs\manual_tests\test_register.md`

Fixes #235

### Type of Change:
<!--**Delete irrelevant options.**-->
- Testing
- Documentation


### How Has This Been Tested?
<!--Describe the tests you ran to verify your changes. Provide instructions or GIFs so we can reproduce them. List any relevant details for your test.-->


### Checklist:
<!--**Delete irrelevant options.**-->

- [x] My PR follows the style guidelines of this project
- [x] I have performed a self-review of my own code or materials
- [x] I have commented my code or provided relevant documentation, particularly in hard-to-understand areas
- [x] I have made corresponding changes to the documentation
- [ ] Any dependent changes have been merged


**Code/Quality Assurance Only**
- [x] My changes generate no new warnings
- [ ] My PR currently breaks something (fix or feature that would cause existing functionality to not work as expected)
- [x] I have added tests that prove my fix is effective or that my feature works
- [ ] New and existing unit tests pass locally with my changes
- [ ] Any dependent changes have been published in downstream modules

Aryamanz29 committed:
test : Manual testing POST /register

devkapilbansal commented:
@Aryamanz29 ensure all test cases are covered and please use link to the image instead of adding images in the repository.

For this you can upload your images in comment copy the URL and use there

Aryamanz29 commented:
@devkapilbansal  Updates : 
- Added some more tests.
- Removed test ss directory.
-  One more thing I've noticed while testing that a new user created when I filling an email field like `xyz@xyz.com123`, See this is an invalid email address, but we get a positive result, instead of getting `400 BAD REQUEST` 

![Screenshot (171)](https://user-images.githubusercontent.com/56113566/120226598-24cf8f80-c265-11eb-98ef-30e930103c88.png)

![Screenshot (172)](https://user-images.githubusercontent.com/56113566/120226603-26995300-c265-11eb-8f6b-1dd4880d83d4.png)

mtreacy002 commented on docs/manual_tests/test_register.md:
```diff
@@ -0,0 +1,112 @@
+
+![post-register-positive](https://user-images.githubusercontent.com/56113566/120222706-01edad00-c25e-11eb-96b0-d0a6d62534e8.gif)
+
+- **Expected Result** : A new user should be created.
```
Since you've shown 3 different examples (this and the following 2 tests) of when a new user should be created based on the combinations of `need mentoring` and `available to mentor` attributes, you can put these 3 under the `Positive tests` section.

mtreacy002 commented on docs/manual_tests/test_register.md:
```diff
@@ -0,0 +1,112 @@
+
+![register-negative-6](https://user-images.githubusercontent.com/56113566/120224858-dc62a280-c261-11eb-84dd-7d1014b43b7e.png)
+
+- **Expected Result** : A new user shouldn't be created.
```
This and the following tests can be put under the `Negative tests` section then give each tests subtitle based on the type of test being done, e.g. `If username already exist, etc`.

mtreacy002 reviewed (changes requested):
@Aryamanz29 , thank you for your work on the issue. I suggest that you can have the tests put under the following structure:
* Test Description: /register [method = POST]
    * Positive tests
        * subsections.... 
     * Negative tests 
        * subsections.... 

Hope this makes sense 😉

mtreacy002 commented:
@aryaman, can we please get an update on this?

Aryamanz29 committed:
test : register / Done Requested Changes

Aryamanz29 commented:
> @Aryaman, can we please get an update on this?

Sorry @mtreacy002  I forgot about this PR 😅,  Done requested changes ✔, Ready for review ✌

mtreacy002 reviewed (approved):
@Aryamanz29 , this looks good to me now 👍.
Thank you for your contribution to the project 🎉🎉🎉. Since this is a documentation pr, no testing is needed. Will approve and merge this now 😉.
